ui/src/project.ts

```diff
@@ -232,21 +271,19 @@ export const extractName = (repoPath: string) =>
   repoPath.split("/").slice(-1)[0];
 
 const fetchBranches = async (path: string) => {
-  // Revert to defaults whenever the path changes in case this query fails
-  // or the user clicks cancel in the directory selection dialog.
-  localState.set("");
+  fetchLocalState({ path });
+
   localStateError.set("");
   defaultBranch.set(DEFAULT_BRANCH_FOR_NEW_PROJECTS);
 
   // This is just a safe guard. Since the validations on the constraints are
   // executed first, an empty path should not make it this far.
-  if (path === "") {
+  if (!path.length) {
```

Comment:
Is there no `empty` function or something? :thinking:

Reply:
Not in JS 😜

Reply:
OOf